pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -328,6 +328,12 @@ func (c *Controller) syncService(key string) error {
 		return nil
 	}
 
+	if service.Spec.Type == v1.ServiceTypeExternalName {
+		// services with Type ExternalName receive no endpoints from this controller;
+		// Ref: https://issues.k8s.io/105986
+		return nil
+	}
```

Comment:
If a Service transitions from a different type like ClusterIP to ExternalName, this will not do anything to clean up any existing EndpointSlices. Is that what we want?

Reply:
You are right, we decided not to introduce deletion in this pr, cause we thought it can be controversial 

So yes, ClusterIP -> ExternalName will result in having leftover endpoints and endpointSlices...

Reply:
Makes sense, thanks!